### PR TITLE
[bashible] `kubernetes-api-proxy` configuration is not updated in manual approval mode

### LINKED ISSUE #4443: [bashible] kubernetes-api-proxy not updating in manual approve mode
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.45.3

### Expected Behavior

Master nodes IPs in kubernetes-api-proxy on nodes are always actual.

### Actual Behavior

If NodeGroup is in manual approve mode, master nodes IPs in kubernetes-api-proxy on nodegroup nodes not updating, and in case of master IPs changes nodes will go to NotReady status

### Steps To Reproduce

_No response_

### Additional Information

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by artbin**
## Description

- Update `kubernetes-api-proxy` configuration before waiting for node approval.
- Change weights for `Bashible` steps.

## Why do we need it, and what problem does it solve?

If `NodeGroup` is in manual approval mode and a node from this `NodeGroup` is waiting for node approval, and 
if the list of IP addresses of master nodes changes, then node will go to `NotReady` status. To solve this problem, 
we need to update the list of IP addresses of master nodes in `kubernetes-api-proxy` configuration before waiting for node approval.

Close #4443

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: bashible
type: fix
summary: Add a node approval annotations step.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by artbin** [comment]
Manual testing
==============

Master node command:
```bash
kubectl create -f - <<EOF
apiVersion: deckhouse.io/v1
kind: NodeGroup
metadata:
  name: manual-approval-worker
spec:
  nodeType: CloudEphemeral
  cloudInstances:
    minPerZone: 1
    maxPerZone: 1
    classReference:
      kind: OpenStackInstanceClass
      name: worker
  disruptions:
    approvalMode: Manual
EOF
```
Output:
```
nodegroup.deckhouse.io/manual-approval-worker created
```

Worker node command:
```bash
ls -1 /var/lib/bashible/bundle_steps/ > steps1.txt
```

Master node command:
```bash
kubectl -n d8-system set image deployment/deckhouse deckhouse=dev-registry.deckhouse.io/sys/deckhouse-oss:pr5047
```
Output:
```
deployment.apps/deckhouse image updated
```

Worker node command:
```bash
journalctl -fu bashible
```
Output:
```
Jun 28 09:42:44 dev-manual-approval-worker-b5a33d15-79b5c-d5488 systemd[1]: Started Bashible service.
Jun 28 09:42:50 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:42:50 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/000_discover_node_ip.sh
Jun 28 09:42:50 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/001_configure_kubernetes_api_proxy.sh
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/001_setting_approval_annotations.sh
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9156]: Setting update.node.deckhouse.io/waiting-for-approval= annotation on our Node...
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9193]: node/dev-manual-approval-worker-b5a33d15-79b5c-d5488 annotated
Jun 28 09:42:51 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9156]: Waiting for update.node.deckhouse.io/approved= annotation on our Node...
Jun 28 09:42:52 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9154]: Steps are waiting for approval to start.
Jun 28 09:42:52 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9154]: Note: Deckhouse is performing a rolling update. If you want to force an update, use the following command.
Jun 28 09:42:52 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9154]: kubectl annotate node dev-manual-approval-worker-b5a33d15-79b5c-d5488 update.node.deckhouse.io/approved=
Jun 28 09:42:52 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9154]: Retry in 10sec...
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/002_add_system_proxy.sh
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/002_install_ca_certificates.sh
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9311]: bb-apt [INFO] Updating apt cache
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9347]: Hit:1 http://nova.clouds.archive.ubuntu.com/ubuntu jammy InRelease
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9347]: Hit:2 http://nova.clouds.archive.ubuntu.com/ubuntu jammy-updates InRelease
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9347]: Hit:3 http://nova.clouds.archive.ubuntu.com/ubuntu jammy-backports InRelease
Jun 28 09:43:02 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9347]: Hit:4 http://security.ubuntu.com/ubuntu jammy-security InRelease
Jun 28 09:43:04 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9347]: Reading package lists...
Jun 28 09:43:04 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9311]: bb-apt [INFO] Installing packages 'ca-certificates'
Jun 28 09:43:04 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9311]: WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
Jun 28 09:43:04 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9607]: Reading package lists...
Jun 28 09:43:04 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9607]: Building dependency tree...
Jun 28 09:43:04 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9607]: Reading state information...
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9607]: ca-certificates is already the newest version (20230311ubuntu0.22.04.1).
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9607]: 0 upgraded, 0 newly installed, 0 to remove and 181 not upgraded.
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[9664]: ca-certificates was already set on hold.
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/002_migrate_containerd_to_cgroupfs.sh
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/003_add_deckhouse_path.sh
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/003_disable_rp_filter.sh
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/003_disable_swap.sh
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/003_install_mandatory_packages.sh
Jun 28 09:43:05 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/003_install_mandatory_packages_for_cloud_nodes.sh
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/003_set_short_hostname.sh
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/004_integrate_kubernetes_data_device.sh
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/004_load_mandatory_modules.sh
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/004_resize_partitions.sh
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: ===
Jun 28 09:43:06 dev-manual-approval-worker-b5a33d15-79b5c-d5488 bashible.sh[8974]: === Step: /var/lib/bashible/bundle_steps/010_disable_unattended_upgrades.sh
```

Worker node command:
```bash
ls -1 /var/lib/bashible/bundle_steps/ > steps2.txt
```

Worker node command:
```bash
diff steps1.txt steps2.txt
```

Output:
```
2,15c2,13
< 000_discover_node_ip.sh
< 001_configure_kubernetes_api_proxy.sh
< 001_setting_approval_annotations.sh
< 002_add_system_proxy.sh
< 002_install_ca_certificates.sh
< 002_migrate_containerd_to_cgroupfs.sh
< 003_add_deckhouse_path.sh
< 003_disable_rp_filter.sh
< 003_disable_swap.sh
< 003_install_mandatory_packages.sh
< 003_install_mandatory_packages_for_cloud_nodes.sh
< 003_set_short_hostname.sh
< 004_integrate_kubernetes_data_device.sh
< 004_load_mandatory_modules.sh
---
> 000_add_system_proxy.sh
> 000_install_ca_certificates.sh
> 000_migrate_containerd_to_cgroupfs.sh
> 001_add_deckhouse_path.sh
> 001_disable_rp_filter.sh
> 001_disable_swap.sh
> 001_install_mandatory_packages.sh
> 001_install_mandatory_packages_for_cloud_nodes.sh
> 001_set_short_hostname.sh
> 002_integrate_kubernetes_data_device.sh
> 002_load_mandatory_modules.sh
> 003_discover_node_ip.sh
30a29
> 052_configure_kubernetes_api_proxy.sh
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

File: candi/bashible/common-steps/node-group/001_setting_approval_annotations.sh.tpl
Comment (RomanenkoDenys) [inline candi/bashible/common-steps/node-group/001_setting_approval_annotations.sh.tpl:1]: i think name should be waiting_approval_annotations.sh.tpl

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/static'...

**Comment by artbin** [comment]
/e2e/abort azure;Standard;containerd;1.23 5474493159 5474493159-1-con-1-23 /sys/deckhouse-oss/install:pr5047 azureuser@20.86.148.178

**Comment by artbin** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.23 5474493339 5474493339-1-con-1-23 /sys/deckhouse-oss/install:pr5047 cloud-user@158.160.107.198

**Comment by artbin** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.23 5474493339 5474493339-1-con-1-23 /sys/deckhouse-oss/install:pr5047 cloud-user@158.160.107.198

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/static'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/vsphere'...

**Comment by RomanenkoDenys** [comment]
/e2e/abort azure;Standard;containerd;1.23 5475969716 5475969716-1-con-1-23 /sys/deckhouse-oss/install:pr5047 azureuser@68.219.177.222

**Comment by RomanenkoDenys** [comment]
/e2e/abort azure;Standard;containerd;1.23 5475969716 5475969716-1-con-1-23 /sys/deckhouse-oss/install:pr5047 azureuser@68.219.177.222

**Comment by name212** [comment]
/e2e/abort azure;Standard;containerd;1.23 5475969716 5475969716-1-con-1-23 /sys/deckhouse-oss/install:pr5047 azureuser@68.219.177.222

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/azure'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Pull request merged**